pkg/cloud/openstack/openstack.go

```diff
@@ -80,3 +83,59 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
 
 	return assets, nil
 }
+
+// CloudConfigTransformer implements the cloudConfigTransformer. It uses the source ConfigMap and
+// to create a new config that adheres to the configuration for CCM
+// It returns an error if the platform is not OpenStackPlatformType.
+func CloudConfigTransformer(source string, infra *configv1.Infrastructure) (string, error) {
+	if !(infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Type == configv1.OpenStackPlatformType) {
+		return "", fmt.Errorf("invalid platform, expected to be %s", configv1.OpenStackPlatformType)
+	}
+
+	cfg, err := ini.Load([]byte(source))
+	if err != nil {
+		return "", fmt.Errorf("failed to read the cloud.conf: %w", err)
+	}
+
+	global, _ := cfg.GetSection("Global")
+	if global != nil {
+		// Remove the legacy keys, once we ensure they're not overridden
+		for key, value := range map[string]string{
+			"secret-name":      "openstack-credentials",
+			"secret-namespace": "kube-system",
+			"kubeconfig-path":  "",
+		} {
+			if global.Key(key).String() != value {
+				return "", fmt.Errorf("'[Global] %s' is set to a non-default value", key)
+			}
+			global.DeleteKey(key)
+		}
+	} else {
+		// Section doesn't exist, ergo no validation to concern ourselves with
```

Comment:
For a moment I was wondering if this should be an error case as there's absolutely no chance the original cloud.conf would have been functional since it had no creds to communicate with OpenStack.
But I understand the intention and agree it's nicer to the user to recover from this (unlikely) situation where the cloud.conf doesn't have a Global section.